### PR TITLE
refs #9881 - discovery_rules works when organizations and locations are disabled

### PR DESCRIPTION
**Pull request by orrabin**
This PR fixed @domcleal's comments on https://github.com/theforeman/foreman_discovery/pull/172

@lzap - this isn't completely ready yet. I opened this so we can test using it and decide what to do when discovery_rules doesn't have a organization/location (when they are enabled and disabled)? when the host doesn't have them? when neither have them?
I will write tests according to the conclusions to these questions.

### CODE REVIEW

**Comment by orrabin** [comment]
[test]

**Comment by domcleal** [comment]
The failures currently are due to #9940.

**Comment by orrabin** [comment]
@domcleal thanks, I wont run the tests again for now

File: app/controllers/concerns/foreman/controller/discovered_extensions.rb [hunk 1]
```diff
@@ -29,7 +29,17 @@ def find_discovery_rule host
...
+    end
+
+    def validate_rule_by_taxonomy rule, host
+      if SETTINGS[:organizations_enabled]
+        return false unless rule.organizations.include?(host.organization)
+      end
+      if SETTINGS[:locations_enabled]
+        return false unless rule.locations.include?(host.location)
+      end
+      return true
```
Comment (lzap) [inline app/controllers/concerns/foreman/controller/discovered_extensions.rb]: Should be only `true` otherwise the cop will scream out

**Comment by lzap** [comment]
This is fine, if user enable org/loc she needs to reboot hosts. We can document that.

**Comment by lzap** [comment]
Hmmm #9940 looks like resolved, [test]

**Comment by lzap** [comment]
Ok it works just fine, the edge case is when user discover hosts with o/l disabled, then enable it. Auto provision won't trigger rules because discovered hosts has no o/l set. Manual provisioning will not work either because o/l dropdowns are disabled (limitation of editing existing hosts with o/l enabled - RFE in the Core). Merged as 47ecc19 thanks.

**Comment by domcleal** [comment]
@lzap sounds like you're describing http://projects.theforeman.org/issues/4426
